sdk/search/search-documents/test/public/node/searchClient.spec.ts

```diff
@@ -14,25 +14,64 @@ import {
   KnownQueryLanguage,
   KnownSpeller,
   SearchClient,
+  SearchIndex,
   SearchIndexClient,
   SelectFields,
 } from "../../../src";
 import { SearchFieldArray, SelectArray } from "../../../src/indexModels";
-import { delay, serviceVersions } from "../../../src/serviceUtils";
+import { defaultServiceVersion } from "../../../src/serviceUtils";
 import { Hotel } from "../utils/interfaces";
 import { createClients } from "../utils/recordedClient";
 import { createIndex, createRandomIndexName, populateIndex, WAIT_TIME } from "../utils/setup";
 
-versionsToTest(serviceVersions, {}, (serviceVersion, onVersions) => {
-  onVersions({ minVer: "2023-11-01" }).describe("SearchClient tests", function (this: Suite) {
+describe("SearchClient", function (this: Suite) {
+  this.timeout(20_000);
+
+  describe("constructor", function () {
+    const credential = new AzureKeyCredential("key");
+
+    describe("Passing serviceVersion", () => {
+      const [correctServiceVersion, incorrectServiceVersion] = ["correct", "incorrect"];
+      it("supports passing serviceVersion", () => {
+        const client = new SearchClient<Hotel>("", "", credential, {
+          serviceVersion: correctServiceVersion,
+        });
+        assert.equal(correctServiceVersion, client.serviceVersion);
+        assert.equal(correctServiceVersion, client.apiVersion);
+      });
+
+      it("supports passing the deprecated apiVersion", () => {
+        const client = new SearchClient<Hotel>("", "", credential, {
+          apiVersion: correctServiceVersion,
+        });
+        assert.equal(correctServiceVersion, client.serviceVersion);
+        assert.equal(correctServiceVersion, client.apiVersion);
+      });
+
+      it("prioritizes `serviceVersion` over `apiVersion", () => {
```

Comment:
It's a common miss when deprecating a field for another field (wha thappens if I pass both) - glad you're thinking and testing this!